src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/TypeAndMethod.cs

```diff
@@ -48,7 +51,8 @@ public override int GetHashCode()
                 unchecked(Method.GetHashCode() * 31) ^ 
                 (IsUnboxingStub ? -0x80000000 : 0) ^ 
                 (IsInstantiatingStub ? 0x40000000 : 0) ^
-                (IsPrecodeImportRequired ? 0x20000000 : 0);
+                (IsPrecodeImportRequired ? 0x20000000 : 0) ^
+                (SignatureContext.GetHashCode() * 23);
```

Comment:
>SignatureContext [](start = 17, length = 16)

Can this ever be null?

Reply:
I believe that the signature context may never be null. The signature context basically represents the token context and as such it gives meaning to the tokens themselves, so to say; without it the signatures are just not meaningful. Please remember our recent discussion related to token context during devirtualization in JIT, especially the large version bubble compilation mode is extremely sensitive to token consistency.